app/src/main/java/com/example/vshcheglov/webshop/ui/MainActivity.kt

```diff
@@ -3,35 +3,55 @@ package com.example.vshcheglov.webshop.ui
 import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
 import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
 import android.view.View
 import android.widget.Toast
 import com.example.vshcheglov.webshop.R
 import com.example.vshcheglov.webshop.data.NetworkService
 import com.example.vshcheglov.webshop.domain.Product
 import com.example.vshcheglov.webshop.ui.adapters.ProductsRecyclerAdapter
+import com.example.vshcheglov.webshop.utils.isNetworkAvailable
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.observers.DisposableSingleObserver
 import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main_primary.*
+import kotlinx.android.synthetic.main.activity_main_error_layout.*
 import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var productsRecyclerAdapter: ProductsRecyclerAdapter
-    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
-    private var productList: List<Product> = mutableListOf()
+    private val compositeDisposable = CompositeDisposable()
+    private var productList = mutableListOf<Product>()
+
+    private var isActivityActive = true
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        productsRecyclerAdapter = ProductsRecyclerAdapter(productList)
-        with(productsRecyclerView) {
-            layoutManager = LinearLayoutManager(this@MainActivity)
-            adapter = productsRecyclerAdapter
+        loadProductsIfInternetAvailable()
```

Comment:
too complicated function name... you just "setupProductsList" - that's it, Internet check - this is inside the function